views/components/ShareCharts.jsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react'
+import Select from '@material-ui/core/Select'
+import Dialog from '@material-ui/core/Dialog'
+import DialogActions from '@material-ui/core/DialogActions'
+import DialogContent from '@material-ui/core/DialogContent'
+import DialogTitle from '@material-ui/core/DialogTitle'
+import Chip from '@material-ui/core/Chip'
+import MenuItem from '@material-ui/core/MenuItem'
+import Typography from '@material-ui/core/Typography'
+import Button from '@material-ui/core/Button'
+import Input from '@material-ui/core/Input'
+
+const menuProps = {
+  PaperProps: {
+    style: {
+      marginTop: '50px',
+    },
+  },
+}
+
+const ShareChart = ({
+  open,
+  chart,
+  userList,
+  handleChange,
+  handleDelete,
+  shareWithUsers,
+  closeDialog,
+}) => {
+  return (
+    <Dialog open={open} onClose={closeDialog} fullScreen={true}>
+      <DialogTitle
+        id='alert-dialog-title'
+        disableTypography={true}
+        style={{
+          backgroundColor: 'rgba(0,0,0,0.7)',
+        }}
+      >
+        <Typography
+          variant='title'
+          style={{
+            color: '#ffffff',
+          }}
+        >
+          Share your chart
+        </Typography>
+      </DialogTitle>
+      <DialogContent
+        style={{
+          padding: '24px',
+          overflowY: 'visible',
+        }}
+      >
+        <Select
+          multiple
+          value={chart.sharedWith || []}
+          onChange={handleChange}
+          input={<Input id='select-multiple' />}
+          MenuProps={menuProps}
+          fullWidth
+          renderValue={(selected) => (
+            <>
+              {selected.map((value) => (
+                <Chip
+                  key={value}
+                  label={value}
+                  onDelete={() => handleDelete(value)}
+                  value={value}
+                  onMouseDown={(event) => {
+                    event.stopPropagation()
+                  }}
```

Comment:
why do we need to handle mouseDown events?

Reply:
The Material v3 events are a mouseDownEvent https://github.com/mui/material-ui/blob/79ebf71d31cca2c73143891960ea54ec1132bdeb/packages/material-ui/src/Select/SelectInput.js#L374.
There was an issue with some versions of material where a Chip in a multi select did not work and event.stopPropagation() was the workaround. I think it's been fixed in more recent versions. 
https://github.com/mui/material-ui/issues/18658#issuecomment-799453268